### PR TITLE
Add Ubuntu 22.04

### LINKED ISSUE #61995: Add Ubuntu 22.04 to CI
https://github.com/saltstack/salt-jenkins/pull/1766

### PR DESCRIPTION
**Pull request by krionbsd**
### What does this PR do?
Add Ubuntu 22.04

Fixes: https://github.com/saltstack/salt/issues/61995 https://github.com/saltstack/salt/issues/61779

### CODE REVIEW

**Comment by MKLeb** [comment]
re-run all

**Comment by MKLeb** [comment]
re-run full pr-ubuntu-2204

**Comment by MKLeb** [comment]
re-run full pr-ubuntu-2204

**Comment by MKLeb** [comment]
re-run full pr-ubuntu-2204

File: tests/pytests/integration/ssh/test_py_versions.py [hunk 1]
```diff
@@ -135,6 +135,7 @@ def test_py36_target(salt_ssh_cli):
     Test that a python >3.6 master can salt ssh to a <3.6 target
     """
     ret = salt_ssh_cli.run("test.ping", minion_tgt="pyvertest")
+    if "kex_exchange_identification" in ret.stdout:
```
Comment (Ch3LL) [inline tests/pytests/integration/ssh/test_py_versions.py:138]: Is this occurring each time on ubuntu 22.04?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/integration/ssh/test_py_versions.py (hunk 1 above)
Comment (MKLeb) [reply tests/pytests/integration/ssh/test_py_versions.py:138]: Not each time, it seems to happen randomly

File: requirements/static/ci/py3.10/linux-crypto.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #
 #    pip-compile --output-file=requirements/static/ci/py3.10/linux-crypto.txt --pip-args='--constraint=requirements/static/ci/py3.10/linux.txt' requirements/static/ci/crypto.in
 #
-m2crypto==0.35.2
+m2crypto==0.38.0
```
[the remark this answers is not in the document]
Comment (MKLeb) [reply requirements/static/ci/py3.10/linux-crypto.txt:7]: 0.35.2 fails to build on python 3.10, switching to 0.38.0 solves that problem

**Comment by MKLeb** [comment]
re-run full pr-ubuntu-2204

**Comment by MKLeb** [comment]
re-run full pr-ubuntu-2204

File: changelog/62100.changed [hunk 3]
```diff
@@ -0,0 +1 @@
+"Sign before ending the testrun in x509.create_certificate"
```
Comment (Ch3LL) [inline changelog/62100.changed:1]: Why are you signging before ending the test run? Normally with test runs you would not want to change behavior.

Comment (MKLeb) [reply changelog/62100.changed:1]: Using m2crypto v0.38.0 causes `tests/integration/states/test_x509.py::test_file_properties_are_updated` to fail with an `Illegal zero content` error.  This solves that problem.  The signed private key from the test run is just returned from `create_certificate` back to `certificate_managed` to affirm that the given certificate (not the one produced by the test run) is valid.  This shouldn't change any behavior, it just moves where the test run ends.  Nothing is added or subtracted from the normal flow of that function, and no new keys are written to the file system.

File: requirements/static/ci/py3.10/linux-crypto.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (MKLeb) [reply requirements/static/ci/py3.10/linux-crypto.txt:7]: Done, will adjust parameters if/when failures arise.

**Comment by Foorack** [comment]
Please also tag Fixes #61799 as #61995 is a duplicate of that issue.

File: requirements/static/ci/py3.10/linux-crypto.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (MKLeb) [reply requirements/static/ci/py3.10/linux-crypto.txt:7]: Can confirm m2crypto==0.38.0 installs on python 3.5-3.10

**Review by Ch3LL — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

File: changelog/62100.changed (hunk 3 above)
Comment (OrangeDog) [reply changelog/62100.changed:1]: Is this going to make #59169 worse?

Or is this only during tests? In which case I don't know why there's a changelog entry for it.

**Pull request merged**